### PR TITLE
Remove or implement unused variables and procs

### PR DESCRIPTION
**Pull request by out-of-phaze**
## Description of changes
Removes a bunch of unused variables and procs, implements/uses some other ones.

In terms of new features:
- Re-adds the name/job labels to PDAs. At some point I might make this use the labels extension, right now I can't be assed.
- Some optimizations I forgot to nab before

Also includes a few optimizations I forgot to pull in :(

## Why and what will this PR improve
code quality, restoring a feature that got clobbered in the move from device->modcomp PDAs, some missing optimizations oops

## Authorship
Me

## Changelog
:cl:
add: Readds name and job labels to PDAs obtained via job equipment
/:cl:

### CODE REVIEW

**Comment by out-of-phaze** [comment]
I can pull the optimizations out into another PR I was just being lazy. Would also appreciate a closer look at the state machine stuff

File: code/datums/mind/mind.dm [hunk 1]
```diff
@@ -354,9 +354,6 @@
 					if("download")
 						new_objective = new /datum/objective/download
 						new_objective.explanation_text = "Download [target_number] research levels."
-					if("capture")
```
Comment (MistakeNot4892) [inline code/datums/mind/mind.dm:357]: Why are you removing this one?

File: code/game/objects/item_mob_overlay.dm [hunk 2]
```diff
@@ -16,7 +16,7 @@ var/global/list/bodypart_to_slot_lookup_table = list(
 	else
 		has_inventory_icon = FALSE
 	if(. || update_icon)
-		update_icon()
+		queue_icon_update()
```
Comment (MistakeNot4892) [inline code/game/objects/item_mob_overlay.dm]: This is going to flood the queue during atoms init.

Comment (out-of-phaze) [reply code/game/objects/item_mob_overlay.dm]: from what I recall I profiled it and it was a net positive but I can easily drop the commit

File: code/datums/mind/mind.dm (hunk 1 above)
Comment (out-of-phaze) [reply code/datums/mind/mind.dm:357]: what even are "capture points"

Comment (MistakeNot4892) [reply code/datums/mind/mind.dm:357]: They're a function of species rarity iirc. If you're removing this you might as well remove the data theft objective too as neither of them are actually tracked for greentext.

File: code/game/objects/item_mob_overlay.dm (hunk 2 above)
Comment (MistakeNot4892) [reply code/game/objects/item_mob_overlay.dm]: If it doesn't have an impact in profiling then I don't mind but as far as I can tell this is going to make every item queue.

File: code/datums/mind/mind.dm (hunk 1 above)
Comment (out-of-phaze) [reply code/datums/mind/mind.dm:357]: that's still something you can subjectively measure, though? research levels exist, data disks exist. capture points are not mentioned anywhere else in the codebase

**Review by MistakeNot4892 — APPROVED** [review]

**Pull request merged**